### PR TITLE
ENH: support limit keyword in GeoSeries.fillna

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Following the implementation of https://github.com/pandas-dev/pandas/pull/58249

This should fix the dev CI build, because pandas enabled this feature for EAs recently and added tests for it, and those tests were now failing.

### CODE REVIEW

File: geopandas/array.py [hunk 1]
```diff
@@ -1208,30 +1208,6 @@ def take(self, indices, allow_fill=False, fill_value=None):
             result[~shapely.is_valid_input(result)] = None
         return GeometryArray(result, crs=self.crs)
 
-    def _fill(self, idx, value):
```
Comment (jorisvandenbossche) [inline geopandas/array.py:1211]: I merged this helper function in `fillna` itself, because that was the only place where it was actually used

**Comment by jorisvandenbossche** [comment]
At least the dev CI is green now .. ;)

**Comment by jorisvandenbossche** [comment]
And now all builds are green!

**Review by martinfleis — APPROVED** [review]
Is it worth adding a changelog note?

**Pull request merged**